be/src/vec/common/hash_table/hash_table.h

```diff
@@ -245,11 +245,14 @@ struct HashTableGrower {
     /// The state of this structure is enough to get the buffer size of the hash table.
     doris::vectorized::UInt8 size_degree = initial_size_degree;
 
+    size_t _buf_size = 1ULL << initial_size_degree;
+    size_t _max_fill = 1ULL << (initial_size_degree - 1);
+
     /// The size of the hash table in the cells.
-    size_t buf_size() const { return 1ULL << size_degree; }
+    size_t buf_size() const { return _buf_size; }
```

Comment:
if the size_degree is modified in increase_size method. The buffer size should changed but in your code, the buffer size is const.